src/FieldDataPluginFramework.Serialization/JsonConfig.cs

```diff
@@ -334,6 +341,35 @@ private static void ConfigureSpecialParsers()
                     item.VelocityAverageValue = json.GetObject<Measurement>(nameof(item.VelocityAverage))?.Value;
                 });
 
+            Configure(json => new OtherDischargeSection(
+                    json.GetObject<DateTimeInterval>(nameof(OtherDischargeSection.MeasurementPeriod)),
+                    json.Get<string>(nameof(OtherDischargeSection.ChannelName)),
+                    json.GetObject<Measurement>(nameof(OtherDischargeSection.Discharge)),
+                    json.Get<string>(nameof(OtherDischargeSection.DistanceUnitId)),
+                    json.Get<string>(nameof(OtherDischargeSection.MonitoringMethodCode))));
+
+            Configure(json => new EngineeredStructureDischarge(
```

Comment:
This will deserialize the optional nested properties like the head readings or volumetric discharge readings right?

Reply:
Yup, the head readings or volumetric discharge readings are handled automatically. (And I've confirmed that in bench testing too)

This funky logic you see here only lists the exceptions-to-the-rule, ie. when a property does not have both a getter and setter.